tests/phpunit/tests/template.php

```diff
@@ -629,6 +629,43 @@ public function data_extra_args_for_add_settings_section() {
 		);
 	}
 
+	/**
+	 * Tests that `locate_template()` uses the current theme even after switching the theme.
+	 *
+	 * @ticket 18298
+	 *
+	 * @covers ::locate_template
+	 */
+	public function test_locate_template_uses_current_theme() {
+		// Look for parent themes with an index.php template.
+		$themes = array_values(
+			array_filter(
+				wp_get_themes(),
+				static function( $theme ) {
+					if ( $theme->get_stylesheet() !== $theme->get_template() ) {
+						return false;
+					}
+					$php_templates = $theme['Template Files'];
+					return isset( $php_templates['index.php'] );
```

Comment:
@mukeshpanchal27 See the comment above, this is to only consider themes with an `index.php` template (i.e. not a block theme).

For block themes without any PHP templates, the results may differ so that's not something this test could cover.